cli/command/formatter/stats.go

```diff
@@ -145,8 +145,10 @@ func (c *containerStatsContext) Container() string {
 
 func (c *containerStatsContext) Name() string {
 	c.AddHeader(nameHeader)
-	name := c.s.Name[1:]
-	return name
+	if len(c.s.Name) > 1 {
+		return c.s.Name[1:]
+	}
+	return ""
```

Comment:
There are actually two panic cases:

In case there are only dead containers, `containerStatsContext` has values in certain fields, but not name and ID. (`IsInvalid` is `true`)

In case there is no active containers, the panic is caused by faking of the header:
https://github.com/docker/docker/blob/04f7a03359ac009ad2cd548fa93124a66e880ae5/cli/command/formatter/formatter.go#L74-L78
In this situation, the `containerStatsContext` simply has no value and every field is default empty value. (even `IsInvalid` is `false` because false is default.)

For that we do need to return some value here as `c.s.Name` could be empty.